server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -71,22 +70,17 @@ public MailboxFactory(MailboxManager mailboxManager) {
     private Optional<Mailbox> fromMessageManager(MessageManager messageManager, MailboxSession mailboxSession) throws MailboxException {
         MailboxPath mailboxPath = messageManager.getMailboxPath();
         Optional<Role> role = Role.from(mailboxPath.getName());
-        MessageManager.MetaData mailboxMetaData = getMailboxMetaData(messageManager, mailboxSession);
         return Optional.ofNullable(Mailbox.builder()
                 .id(messageManager.getId())
                 .name(getName(mailboxPath, mailboxSession))
                 .parentId(getParentIdFromMailboxPath(mailboxPath, mailboxSession).orElse(null))
                 .role(role)
-                .unreadMessages(mailboxMetaData.getUnseenCount())
-                .totalMessages(mailboxMetaData.getMessageCount())
+                .unreadMessages(messageManager.getUnseenMessageCount(mailboxSession))
```

Comment:
Because you have part of Metadata the list (arbitrary long) of all RECENT UIDs. This is the core of the problem (and it was not necessarilly a Cassandra design BTW)